### PR TITLE
Ls thing code value picklists should have an option to add a new code value

### LINKED ISSUE #798: Ls thing code value picklists should have an option to add a new code value

### PR DESCRIPTION
**Pull request by brianbolt**
Fixes #798

### CODE REVIEW

**Review by philippcheung — APPROVED** [review]
Lots of cool stuff -- I remember playing in this file :)

File: modules/Components/src/client/ACASFormFields.coffee [hunk 1]
```diff
@@ -412,7 +412,27 @@ class ACASFormLSCodeValueFieldController extends ACASFormAbstractFieldController
...
+		# Default is to automatically save picklist items to the db
+		# Otherwise the controller will need to explicitly tell editable picklist controller
+		# to save the picklist items to the db.  This is because there is a use case to only save
+		# the picklist items to the db when the user clicks the save button on the thing.
+		plOptions.autoSave = true
+		if @options.autoSavePickListItem? && !@options.autoSavePickListItem
+			plOptions.autoSave = false
+
+		plOptions.width = "349px"
+		@pickListController = new EditablePickListSelect2Controller plOptions
```
Comment (anand-schrodinger) [inline modules/Components/src/client/ACASFormFields.coffee:434]: @brianbolt 
`applyFilter` - `@pickListController.removeFilters` complains about 
```
Uncaught TypeError: this.pickListController.removeFilters is not a function
```

**Comment by brianbolt** [comment]
@anand-schrodinger Dang. Sorry about that I will look into it today.  Thought the filtering part of the parent class and didn't realize switching to the select2 controler would break this.

**Comment by anand-schrodinger** [comment]
No worries! I am glad I was working on related logic and came across it. Thank you for looking into it further!

**Pull request merged**